pkg/resource/hyper_parameter_tuning_job/custom_delta.go

```diff
@@ -31,5 +31,29 @@ func customSetDefaults(
 				}
 			}
 		}
+		// TODO: Remove the block below.
+		// The server side default of KeepAlivePeriodInSeconds is nil, when launching a HPO job.
+		// The code generator currently cannot ignore the field path for resourceConfig.KeepAlivePeriodInSeconds
+		// without also ignoring Trainingjob. This block below should be removed once the code generator supports
+		// removing fields like resourceConfig.KeepAlivePeriodInSeconds
+		if ackcompare.IsNotNil(a.ko.Spec.TrainingJobDefinition) && ackcompare.IsNotNil(b.ko.Spec.TrainingJobDefinition) {
+			if ackcompare.IsNotNil(a.ko.Spec.TrainingJobDefinition.ResourceConfig) && ackcompare.IsNotNil(b.ko.Spec.TrainingJobDefinition.ResourceConfig) {
+				if ackcompare.IsNotNil(a.ko.Spec.TrainingJobDefinition.ResourceConfig.KeepAlivePeriodInSeconds) && ackcompare.IsNil(b.ko.Spec.TrainingJobDefinition.ResourceConfig.KeepAlivePeriodInSeconds) {
```

Comment:
shouldnt desired(a) be empty for this condition instead of notNil?

instead of setting a to nil, should we clear the value from b if it is nil?

Reply:
a is supposed to have a value and latest is supposed to be nil